### DISCUSSION TITLE
PostgreSQL unix socket?

### DISCUSSION THREAD

**Discussion by eisengrau** [netbox-community/netbox#5688] [Q&A]
Hello,

I am currently running a PostgreSQL instance on a different host, with a socket path exposed on a shared storage. I would like to install Netbox on another container, however I do not wish to add a new PGSQL instance there as well. Is it possible to configure Netbox to connect to an unix socket for the PGSQL database, instead of a TCP port? I can imagine something like providing /path/to/pgsqldb.sock instead of "localhost:port"

Thank You,

**Comment by a31amit** [comment]
You can pass Unix socket location in HOST SECTION of settings and try

**Comment by candlerb** [comment]
I'm not sure what you're trying to do.

Unix domain sockets can only be used to communicate between processes on the *same* host.  Putting the socket path on shared storage doesn't alter that.  If Postgres and Netbox are running in different hosts (or different VMs on the same host), then you'll need to use a TCP socket to connect them.

**Reply by eisengrau** [reply]
Hello,

I am running the services in LXC containers, not VMs, so basically these are on the same host, but separated in their own chroot file system, like for example:

/mnt/mystorage/pgsql-container-01/
/mnt/mystorage/pgsql-container-02/
/mnt/mystorage/netbox-container-01/
...
/mnt/sharedstorage/sockets/
...

I can mount /mnt/sharedstorage/sockets/ as /sockets in every relevant container. It will act like a symbolic link basically.

Where ./pgsql-container-01/ is the root directory of the container with a minimal debian and database service running, basically the container identifies it as '/'. I can set the socket path within the configuration to UNIX socket /sockets/pgsql01.sock

./sharedstorage/ contains shared resources, and I can like mount this namespace 'in' the container, and it would be like  /mnt/mystorage/netbox-container-01/sockets/pgsql01.sock

OK, I know that the usual TCP/IP sockets would just work, and I shouldn't bother with these really, also the overhead is negligeable. I was just curios if netbox will be able to listen to such configuration.

**Reply by candlerb** [reply]
Ah, that makes sense: then these containers are on the same host / same OS.

What you need to do is to configure Django to talk to Postgres via a unix domain socket.  Putting the socket path in the HOST parameter should do this, as already suggested by @a31amit. See:
https://docs.djangoproject.com/en/3.1/ref/settings/#host
https://code.djangoproject.com/ticket/21296
